MyArrayList.java

```diff
@@ -1 +1,110 @@
+package com.company;
 
+public class MyArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private int position = 0;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private StringBuilder stringBuilder = new StringBuilder();
+    private int size = array.length;
+
+    @Override
+    public void add(T value) {
+        checkForAdd(position);
+        add(value, position);
+    }
+
+    private void checkForAdd(int index) {
+        if (index < 0) {
+            throw new NoSuchFieldError();
+        }
+        if (index >= size) {
+            size += size / 2;
+            array = newArray();
+        }
+    }
+
+    private T[] newArray() {
+        Object[] newArray = new Object[size];
+        System.arraycopy(array, 0, newArray, 0, position);
+        return (T[]) newArray;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        checkForAdd(index);
+        System.arraycopy(array, index, array, index + 1, size - index - 1);
+        array[index] = value;
+        position++;
+    }
+
+    @Override
+    public void addAll(List<T> list) {
```

Comment:
we need increase position after adding the elements

Reply:
Yes